plugin/evm/limit_order.go

```diff
@@ -160,50 +121,68 @@ func parseTx(orderBookABI abi.ABI, memoryDb limitorders.InMemoryDatabase, backen
 			limitOrder := limitorders.LimitOrder{
 				PositionType:      positionType,
 				UserAddress:       order.Trader.Hash().String(),
-				BaseAssetQuantity: int(order.BaseAssetQuantity.Uint64()),
+				BaseAssetQuantity: int(order.BaseAssetQuantity.Int64()),
 				Price:             price,
 				Salt:              order.Salt.String(),
+				Status:            "unfulfilled",
 				Signature:         signature,
 				RawOrder:          in["order"],
 				RawSignature:      in["signature"],
 			}
 			memoryDb.Add(limitOrder)
+			matchLimitOrderAgainstStoredLimitOrders(txPool, orderBookABI, memoryDb, limitOrder)
 		}
-		if m.Name == "executeTestOrder" {
-			log.Info("##### in ParseTx", "executeTestOrder tx hash", tx.Hash().String())
-			go pollForReceipt(backend, tx.Hash())
-			signature := in["signature"].([]byte)
-			memoryDb.Delete(signature)
+		if m.Name == "executeMatchedOrders" {
+			signature1 := in["signature1"].([]byte)
+			memoryDb.Delete(signature1)
+			signature2 := in["signature2"].([]byte)
+			memoryDb.Delete(signature2)
 		}
 	}
 }
 
-func pollForReceipt(backend eth.EthAPIBackend, txHash common.Hash) {
-	for i := 0; i < 10; i++ {
-		receipt := getTxReceipt(backend, txHash)
-		if receipt != nil {
-			log.Info("receipt found", "tx", txHash.String(), "receipt", receipt)
-			return
+func matchLimitOrderAgainstStoredLimitOrders(txPool *core.TxPool, orderBookABI abi.ABI, memoryDb limitorders.InMemoryDatabase, limitOrder limitorders.LimitOrder) {
+	oppositePositionType := getOppositePositionType(limitOrder.PositionType)
+	if oppositePositionType == "" {
```

Comment:
this should be an error scenario, right? maybe log it as error

Reply:
hmm i am thinking positionType is something we assign in code based on baseAssetQuantity. So this should not happen. This should not happen. I will review it 🤔

Reply:
hmm still there should be a check on zero baseAssetQuantity. Let me add a check for that.  The check for that should be in contract as well but i will add it here as well. Will log it and not store the order. What do you think ?

Reply:
Yep, sounds good. It will never happen as long as contract has that check too